nmdc_schema/migrators/migrator_from_10_3_0_to_10_4_0.py

```diff
@@ -0,0 +1,33 @@
+from nmdc_schema.migrators.migrator_base import MigratorBase
+
+
+class Migrator(MigratorBase):
+    r"""Migrates a database between two schemas."""
+
+    _from_version = "10.3.0"
+    _to_version = "10.4.0"
+
+    def upgrade(self) -> None:
+        r"""Migrates the database from conforming to the original schema, to conforming to the new schema."""
+
+        self.adapter.process_each_document(
+            "nom_analysis_activity_set", [self.update_nom_analysis_activity_id]
+        )
+
+    def update_nom_analysis_activity_id(self, nom_analysis_activity: dict) -> dict:
+        r"""
+        Updates a nom_analysis_activity's id to contain a version number (.1),
+        if it has no version number.
+
+        >>> m = Migrator()
+        >>> m.update_nom_analysis_activity_id({'id': '1'}) # test: adds version number to id
+        {'id': '1.1'}
```

Comment:
Will you add an example for where the document's ID already ends with `.1`?

```suggestion
        {'id': '1.1'}
        >>> m.update_nom_analysis_activity_id({'id': '1.1'}) # test: no change
        {'id': '1.1'}
```

Reply:
I'd also like to see a counterexample where the document's ID contains a dot that is not at the end. That would serve as documentation about the migrator's behavior for such a corner case.

For example:

```py
>>> m.update_nom_analysis_activity_id({'id': 'nmdc:foo-12.3-potato'}) # test: no change
{'id': 'nmdc:foo-12.3-potato'}
```